### PR TITLE
Added jtreg-buffer external suite

### PR DESCRIPTION
**Pull request by judovana**
This PR is adding our buffer of not-yet-if-ever upstreamed jtreg tests, as another external testsuite.
It do pass locally, however in grinder it fails https://ci.adoptium.net/view/Test_grinder/job/Grinder/9229/console  `Error: cannot find the following tests: jtreg-buffer (note: group target such as sanity is not accepted inside testList)`  So i guess there is somwehre some another protection.

Except that issue,  for which I do not have cure, few questions
 - external tests do not run with podman-docker docekr wrapper. The traces are quite cryptic, is this already tracked? 
   - Is there generic/general opinion on moving to root-less container engine (thus podman)?
 - on grinder jobs, the external tests run as root? I had not found a trace of change of user in logs, so am wondering
   - local root-less run proiperly failed for me
 -  is there any way how to copy any testsuite results from container to host?
   - the tap file is correctly on host, but the suite itself is generating jtrfiles and jtr.xml which I would llike to be able to archive on demand if there occure issues.

### CODE REVIEW

**Comment by judovana** [comment]
@sophia-guo  maybe you can provide feedback please?

**Comment by karianna** [comment]
@judovana You should run a spell and grammar checker over the comments

**Comment by smlambert** [comment]
PRs to add new tests or features should have an issue that they relate to, as that is what we typically do at the project to socialize new features.  

I have created this issue, where it would say what the new tests do, why they should be added into a certain level, etc (noting we have the habit of adding new tests into the dev level to monitor how they run before 'graduating' tests into other levels).

**Comment by smlambert** [comment]
https://ci.adoptium.net/view/Test_grinder/job/Grinder/9354 to fix the TARGET which needs to match the name in the playlist.xml file (not the name of the directory)

Also will want to address the criteria/questions listed in #5195 ahead of bringing in new tests.  We will add this to the contributing new tests documentation going forward, so future contributors have awareness.

File: external/jtreg-buffer/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# External jtreg-buffer Tests
+
+jtreg-buffer are tests, which under normal circumstances would end in OpenJDK regression test repositories, but they did not.
```
Comment (karianna) [inline external/jtreg-buffer/README.md:3]:
```suggestion
jtreg-buffer are tests, which ideally should end in OpenJDK regression test repositories.
```

File: external/jtreg-buffer/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+However, original authors disappeared, upstream did not accepted the change, or there was not proper time, or the author was simply lazy.
```
Comment (karianna) [inline external/jtreg-buffer/README.md:4]:
```suggestion
```

File: external/jtreg-buffer/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Those tests are pulled from https://github.com/rh-openjdk/jtreg-buffer repository. Buffer, like queue for waiting to be upstreamed.
```
Comment (karianna) [inline external/jtreg-buffer/README.md:5]:
```suggestion
These tests are pulled from the https://github.com/rh-openjdk/jtreg-buffer repository.
```

File: external/jtreg-buffer/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+On long run, all those tests should be upstreamed but it is approx 2years of man work and maybe upstreamed will not accept them all.
```
Comment (karianna) [inline external/jtreg-buffer/README.md:6]:
```suggestion
```

File: external/jtreg-buffer/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+If there is such test on  your side, feel free to contribute.
```
Comment (karianna) [inline external/jtreg-buffer/README.md:7]:
```suggestion
```

File: external/jtreg-buffer/test.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+github_url="https://github.com/rh-openjdk/jtreg-buffer.git"
+test_results=""
+tag_version="main"
+ubuntu_packages="git wget tar"
+maven_version="3.8.5"
```
Comment (karianna) [inline external/jtreg-buffer/test.properties:5]: would prefer 3.9.6 if possible.

File: external/jtreg-buffer/test.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+source $(dirname "$0")/test_base_functions.sh
+#Set up Java to be used by the jtreg-buffer test
```
Comment (karianna) [inline external/jtreg-buffer/test.sh:16]:
```suggestion
# Set up Java to be used by the jtreg-buffer test
```

**Review by karianna — CHANGES_REQUESTED** [review]

File: external/jtreg-buffer/test.properties (hunk 6 above)
Comment (judovana) [reply external/jtreg-buffer/test.properties:5]: Actually this do not need maven at all. Will drop that line. If it is mandatory for sute to be picked up, I willreturn it with 3.9.6

**Comment by judovana** [comment]
@karianna  thanx alot for eyball! All should be done as you requested. Need to try that maven in grinder now.
@smlambert  Thanx! I will fix grinder job, and will answer  the issue you created.

Can you please answer the remaining questions?
>     external tests do not run with podman-docker docekr wrapper. The traces are quite cryptic, is this already tracked?
>         Is there generic/general opinion on moving to root-less container engine (thus podman)?

In meantime I had found that no, and created https://github.com/adoptium/aqa-tests/issues/5196 hth

>     on grinder jobs, the external tests run as root? I had not found a trace of change of user in logs, so am wondering

please?

>         local root-less run properly failed for me
>     is there any way how to copy any testsuite results from container to host?

big please.

>     the tap file is correctly on host, but the suite itself is generating jtrfiles and jtr.xml which I would like to be able to archive on demand if there occurs issues.

**Comment by judovana** [comment]
> https://ci.adoptium.net/view/Test_grinder/job/Grinder/9354 to fix the TARGET which needs to match the name in the playlist.xml file (not the name of the directory)

https://ci.adoptium.net/view/Test_grinder/job/Grinder/9360/
 
> Also will want to address the criteria/questions listed in #5195 ahead of bringing in new tests. We will add this to the contributing new tests documentation going forward, so future contributors have awareness.

On it. TYVM!

**Comment by judovana** [comment]
https://ci.adoptium.net/view/Test_grinder/job/Grinder/9363/

**Comment by judovana** [comment]
> https://ci.adoptium.net/view/Test_grinder/job/Grinder/9363/

ok, so maven version is mandatory. The run now correctly passed.  I'm not sure why... I left the target and bulild-list as in original one.. Well Maybe not... Probably harsh morning. Glad it is green.

**Comment by sophia-guo** [comment]
> on grinder jobs, the external tests run as root? I had not found a trace of change of user in logs

External tests are running inside container as root, which can be updated to create and use unprivileged user if root acess is not required by external tests.

**Comment by smlambert** [comment]
Before we go too far with this PR, I think it would be good to think about these jtreg tests as possible ones to add into the openjdk group, given the intent was to eventually upstream them to that project (based on answers in https://github.com/adoptium/aqa-tests/issues/5195).  I do not see a need to run them as part of the external group, they could go into dev.openjdk

**Comment by sophia-guo** [comment]
>    is there any way how to copy any testsuite results from container to host?

To make testsuite results available it have to be copy to specific directory. In test.sh you can copy the testresult to /testResults , update test.properties with `test_results="testResults"` and update command in playlist.xml specify --reportsrc .  Similar to [openliberty-mp-tck](https://github.com/adoptium/aqa-tests/tree/master/external/openliberty-mp-tck)

**Comment by sophia-guo** [comment]
I agree with Shelley.  dev.openjdk might be a better category to go based on https://github.com/adoptium/aqa-tests/issues/5195#issuecomment-2036381675

**Comment by judovana** [comment]
> > is there any way how to copy any testsuite results from container to host?
> 
> To make testsuite results available it have to be copied to specific directory. In test.sh you can copy the test result to /testResults , update test.properties with `test_results="testResults"` and update command in playlist.xml specify `--reportsrc` . Similar to [openliberty-mp-tck](https://github.com/adoptium/aqa-tests/tree/master/external/openliberty-mp-tck)
> 
> This way all *.xml files are visible to junit plugin. Also check both ARCHIVE_TEST_RESULTS and KEEP_REPORTDIR will keep the results even when the tests pass.

Thanx a lot!

as `test_results="testResults"` is property, can it have any value, or is it jsut `testResults` waht is allowed?  Also, what is it relative to?

**Comment by judovana** [comment]
> Before we go too far with this PR, I think it would be good to think about these jtreg tests as possible ones to add into the openjdk group, given the intent was to eventually upstream them to that project (based on answers in #5195). I do not see a need to run them as part of the external group, they could go into dev.openjdk

Cool! Thanx!

I willexperiment a bit  with `test_results="testResults"` then close this one and do a new into `dev.openjdk`

**Pull request closed without merging**